src/_pytest/python_api.py

```diff
@@ -123,7 +123,7 @@ def __eq__(self, actual):
         if not np.isscalar(actual):
             try:
                 actual = np.asarray(actual)
-            except:  # noqa
+            except BaseException:
```

Comment:
as followup, i wonder if Exception would be a better limit

Reply:
`np.asarray` seems to except any and all inputs, so I doubt it can even raise at all, but `BaseException` is definitely inappropriate here. I'll just go ahead and change it to `Exception` in this PR, I think it's safe to do.